src/writer/xapianIndexer.cpp

```diff
@@ -70,7 +70,7 @@ void XapianIndexer::indexingPrelude()
   writableDatabase = Xapian::WritableDatabase(indexPath + ".tmp", Xapian::DB_CREATE_OR_OVERWRITE);
   switch (indexingMode) {
     case IndexingMode::TITLE:
-      writableDatabase.set_metadata("valuesmap", "title:0");
+      writableDatabase.set_metadata("valuesmap", "title:0;value_store:1");
```

Comment:
`value_store` doesn't seem to be a good name. As we store the path of the entry, it would be better to call it "path".

Reply:
I don't really understand why path would be a good name, but in any case we should document in the code each metadata and the meaning of the value.

Reply:
Because a variable[*] name should indicate what it contains. As we store a path, "path" seems a good name.
"value_store" doesn't provide any information. We store a value yes, but it is the purpose of a variable. We never name a variable "variable".

[*] This is not technically a variable, but this is a slot in which we can put a value. The same rule applies when we need to name it.

Reply:
AFAIK this metadata is almost free text, so we should have to do something more explicit than just `path`. In particular if beside we have `title:0` which means? titles are not stored as value? What about `path_stored_as_value`?

Reply:
Ok, I think I need to explain how we store things in the xapian database.
We are storing "document" in the xapian database :
- It contains a data. This is the "main content" of the document. It is intended to be the data we want to retrieve and we should not use it in the query step (for performance reason).
- We can associate some terms with it. This is the classical "indexation" step, analyse the content and generate a index based on the terms.
- We can also add some specific values to a document. This can be used as metadata for the document or be used in the query (to filter, sort, ...)

But in xapian, the value are identified by a integer.
We set the value with `document.add_value(index, value)` (to set the `value` in the slot `index`).
And we can use it in several ways :
- `Xapian::LatLongDistancePostingSource(index, centre, metric, distance)` to add query on the geographic position stored in the slot `index`.
- `Xapian::Enquire::set_sort_by_relevance_then_value(index, false)` to sort by relevance and then by value stored in the slot `index`.
- `Xapian::Enquire::set_collapse_key(index)` to collapse (regroup) results using the value stored in the slot `index`.

The question is what index to use ?
We can hard code this in the code. But then it would be pretty difficult to add/remove/change value when we want to fix/add feature and keep compatibility. We need a way to know what values are stored in the document and in which slot.

The the purpose of the "valuesmap" metadata (which is a database metadata).
It is a string telling what values are stored in the documents and in which slot.
- `"title:0"` means we store the title at slot 0.
- `"title:0;wordcount:1;geo.position:2"` means we store title at slot 0, wordcount in slot 1 and geo.position in slot 2.

This way, when doing the query. We can parse the string to get a map and then use this maps this way :
`Xapian::Enquire::set_collapse_key(valuemap["title"])`. We don't have to hardcode the index, we get it from the database metadata.
We give a name to the value. (It is the exact same purpose of the variable, we give a name to a value instead of using its address)

If somehow, we drop the wordcount because we find it useless, we could create a database without it and have a valuesmap `"title:0;geo.position:1"`. Old libraries (on new zim) would know there is no `wordcount` and that `geo.position` are in slot 1 (instead of 2).
New library (on old zim) would know that the `geo.position` is in slot 2 (and don't care about `wordcount` value).

By setting the value map to `"title:0;path:1"` we are telling the reading library that title is stored in slot 0 and path in slot 1.
`value_store` doesn't provide any useful information.
`path_store_as_value` is providing to much information (by definition, path is stored as value)

---

I just realize your misunderstanding : 0/1 are not boolean telling if the value are stored or not.

Reply:
Thanks @mgautierfr! That clears a lot of stuff. Yes, I think `path` is a more appropriate name for slot # 1 in the titles index. It is explicit about what it actually stores. I'll fix this.